coverage-experiment/src/main/java/nl/tudelft/serg/evosql/experiment/Runner.java

```diff
@@ -16,42 +19,90 @@
+    static final int AMOUNT_QUERIES_ESPOCRM = 40;
+    static final int AMOUNT_QUERIES_SUITECRM = 280;
+
     public static void main(String[] args) {
-        int startIndex = Integer.valueOf(args[0]);
-        int stepSize = Integer.valueOf(args[1]);
-        BufferedReader reader_erpnext = new BufferedReader(new InputStreamReader(
-                Runner.class.getResourceAsStream("sql/erpnext_queries.sql")));
-        Stream<String> erpnext = reader_erpnext.lines();
+        if (args.length < 2) return;
 
-        BufferedReader reader_espocrm = new BufferedReader(new InputStreamReader(
-                Runner.class.getResourceAsStream("sql/espocrm_queries.sql")));
-        Stream<String> espocrm = reader_espocrm.lines();
+        final int startIndex = Integer.valueOf(args[0]);
+        final int stepSize = Integer.valueOf(args[1]);
+        int stopIndex = Integer.MAX_VALUE;
 
-        BufferedReader reader_suitecrm = new BufferedReader(new InputStreamReader(
-                Runner.class.getResourceAsStream("sql/suitecrm_queries.sql")));
-        Stream<String> suitecrm = reader_suitecrm.lines();
+        if (args.length > 2) stopIndex = Integer.valueOf(args[2]);
 
-        try {
-            reader_erpnext.close();
-            reader_espocrm.close();
-            reader_suitecrm.close();
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
+        System.out.printf("Running experiment, starting at %d and stepping %d...", startIndex, stepSize);
+        System.out.println();
 
         QueryReader queryReader = new QueryReader();
-        List<String> allQueries = queryReader.readQueries(erpnext, espocrm, suitecrm);
-
-        // TODO: Make connection data for separate databases...
-        for (int i = startIndex; i < allQueries.size(); i += stepSize) {
-            runForQuery(
-                    allQueries.get(i),
-                    null,
-                    null,
-                    String.valueOf(i)
-            );
+        List<String> allQueries = queryReader.readExperimentQueries();
+
+        stopIndex = Math.min(stopIndex, allQueries.size());
+        System.out.printf("The maximum query that will be executed is %d.", stopIndex - 1);
+        System.out.println();
+
+        ConnectionData connectionDataProd = CONNECTION_DATA_ERPNEXT_PROD;
+        ConnectionData connectionDataTest = CONNECTION_DATA_ERPNEXT_TEST;
+        for (int i = startIndex; i < stopIndex; i += stepSize) {
+            // Sorry for this...
+            if (i >= AMOUNT_QUERIES_ERPNEXT && i < AMOUNT_QUERIES_ERPNEXT + AMOUNT_QUERIES_ESPOCRM) {
+                connectionDataProd = CONNECTION_DATA_ESPOCRM_PROD;
+                connectionDataTest = CONNECTION_DATA_ESPOCRM_TEST;
+            } else if (i >= AMOUNT_QUERIES_ERPNEXT + AMOUNT_QUERIES_ESPOCRM) {
+                connectionDataProd = CONNECTION_DATA_SUITECRM_PROD;
+                connectionDataTest = CONNECTION_DATA_SUITECRM_TEST;
```

Comment:
This is a cluttered mess and I'm not sure whether this is the right approach to re-assign the `ConnectionData` objects. I was thinking that we store the database name string together with the query in the `allQueries` variable or something.

Reply:
Might be useful but not sure by how much. Currently, the `run query` method uses this data structure to figure out the query, and this works now, I'd rather not break it. If you can fix it in a clean way I'm obviously be fine with that.